diff -Nur clutter-gtk-old/clutter-gtk/Makefile.am clutter-gtk/clutter-gtk/Makefile.am
--- clutter-gtk-old/clutter-gtk/Makefile.am	2009-11-07 00:30:33.328247278 +0200
+++ clutter-gtk/clutter-gtk/Makefile.am	2009-11-07 00:35:53.481247340 +0200
@@ -40,7 +40,7 @@
 libclutter_gtk_@CLUTTER_GTK_API_VERSION@_la_LIBADD = $(CLUTTER_LIBS) $(GTK_LIBS)
 libclutter_gtk_@CLUTTER_GTK_API_VERSION@_la_LDFLAGS = $(CLUTTER_LT_LDFLAGS)
 
-cluttergtkheadersdir = $(includedir)/clutter-gtk-1.0/clutter-gtk
+cluttergtkheadersdir = $(includedir)/clutter-1.0/clutter-gtk
 
 # please, keep the list sorted alphabetically
 cluttergtkheaders_HEADERS = \
diff -Nur clutter-gtk-old/clutter-gtk.pc.in clutter-gtk/clutter-gtk.pc.in
--- clutter-gtk-old/clutter-gtk.pc.in	2009-11-07 00:30:33.339245746 +0200
+++ clutter-gtk/clutter-gtk.pc.in	2009-11-07 00:34:54.678248503 +0200
@@ -9,6 +9,6 @@
 Name: clutter-gtk
 Description: GTK+ integration for Clutter
 Version: @VERSION@
 Libs: -L${libdir} -lclutter-gtk-${apiversion}
-Cflags: -I${includedir}/clutter-gtk-${apiversion}
+Cflags: -I${includedir}/clutter-${clutterapiversion}
 Requires: @CLUTTER_GTK_REQUIRES@
