diff -up glade-2.12.2/glade/gbwidgets/gbclist.c.gtk-includes glade-2.12.2/glade/gbwidgets/gbclist.c
--- glade-2.12.2/glade/gbwidgets/gbclist.c.gtk-includes	2007-12-08 17:56:06.000000000 -0500
+++ glade-2.12.2/glade/gbwidgets/gbclist.c	2008-10-01 13:12:37.000000000 -0400
@@ -19,10 +19,7 @@
 
 #include <string.h>
 
-#include <gtk/gtkclist.h>
-#include <gtk/gtkhbox.h>
-#include <gtk/gtkmain.h>
-#include <gtk/gtkspinbutton.h>
+#include <gtk/gtk.h>
 #include "../gb.h"
 
 /* Include the 21x21 icon pixmap for this widget, to be used in the palette */
diff -up glade-2.12.2/glade/glade_keys_dialog.c.gtk-includes glade-2.12.2/glade/glade_keys_dialog.c
--- glade-2.12.2/glade/glade_keys_dialog.c.gtk-includes	2007-12-08 17:56:06.000000000 -0500
+++ glade-2.12.2/glade/glade_keys_dialog.c	2008-10-01 13:12:37.000000000 -0400
@@ -23,11 +23,7 @@
 
 #include <string.h>
 
-#include <gtk/gtkbox.h>
-#include <gtk/gtkbutton.h>
-#include <gtk/gtkclist.h>
-#include <gtk/gtkstock.h>
-#include <gtk/gtkscrolledwindow.h>
+#include <gtk/gtk.h>
 
 #include "gladeconfig.h"
 
diff -up glade-2.12.2/glade/glade_menu_editor.c.gtk-includes glade-2.12.2/glade/glade_menu_editor.c
--- glade-2.12.2/glade/glade_menu_editor.c.gtk-includes	2008-10-01 13:14:30.000000000 -0400
+++ glade-2.12.2/glade/glade_menu_editor.c	2008-10-01 13:14:46.000000000 -0400
@@ -26,32 +26,7 @@
 #include <time.h>
 
 #include <gdk/gdkkeysyms.h>
-#include <gtk/gtkarrow.h>
-#include <gtk/gtkaccellabel.h>
-#include <gtk/gtkclist.h>
-#include <gtk/gtkcombo.h>
-#include <gtk/gtkentry.h>
-#include <gtk/gtkeventbox.h>
-#include <gtk/gtkfilechooserdialog.h>
-#include <gtk/gtkframe.h>
-#include <gtk/gtkhbox.h>
-#include <gtk/gtkhbbox.h>
-#include <gtk/gtkhseparator.h>
-#include <gtk/gtkiconfactory.h>
-#include <gtk/gtkimage.h>
-#include <gtk/gtkimagemenuitem.h>
-#include <gtk/gtklist.h>
-#include <gtk/gtkmain.h>
-#include <gtk/gtkmenu.h>
-#include <gtk/gtkmenubar.h>
-#include <gtk/gtkradiobutton.h>
-#include <gtk/gtkradiomenuitem.h>
-#include <gtk/gtkscrolledwindow.h>
-#include <gtk/gtkseparatormenuitem.h>
-#include <gtk/gtkstock.h>
-#include <gtk/gtktable.h>
-#include <gtk/gtktearoffmenuitem.h>
-#include <gtk/gtkvbox.h>
+#include <gtk/gtk.h>
 
 #include "gladeconfig.h"
 
diff -up glade-2.12.2/glade/glade_project_view.c.gtk-includes glade-2.12.2/glade/glade_project_view.c
diff -up glade-2.12.2/glade/glade_project_view.h.gtk-includes glade-2.12.2/glade/glade_project_view.h
--- glade-2.12.2/glade/glade_project_view.h.gtk-includes	2008-10-01 13:16:07.000000000 -0400
+++ glade-2.12.2/glade/glade_project_view.h	2008-10-01 13:16:18.000000000 -0400
@@ -18,7 +18,7 @@
 #ifndef GLADE_PROJECT_VIEW_H
 #define GLADE_PROJECT_VIEW_H
 
-#include <gtk/gtkclist.h>
+#include <gtk/gtk.h>
 
 #include "glade_project.h"
 