--- src/drivers/driver_nl80211.c.orig	2012-05-10 00:56:09.000000000 +0300
+++ src/drivers/driver_nl80211.c	2012-10-14 14:54:11.858031717 +0300
@@ -22,9 +22,9 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <net/if.h>
-#include <netlink/genl/genl.h>
-#include <netlink/genl/family.h>
-#include <netlink/genl/ctrl.h>
+#include <libnl3/netlink/genl/genl.h>
+#include <libnl3/netlink/genl/family.h>
+#include <libnl3/netlink/genl/ctrl.h>
 #include <linux/rtnetlink.h>
 #include <netpacket/packet.h>
 #include <linux/filter.h>
@@ -36,7 +36,7 @@
 #include "common/ieee802_11_defs.h"
 #include "common/ieee802_11_common.h"
 #include "l2_packet/l2_packet.h"
-#include "netlink.h"
+#include "libnl3/netlink/netlink.h"
 #include "linux_ioctl.h"
 #include "radiotap.h"
 #include "radiotap_iter.h"
