diff -ur xpisi-2.4/pisi.conf-i686 pisi-2.4/pisi.conf-i686
--- pisi.conf-i686	2012-10-01 22:51:06.000000000 +0300
+++ pisi.conf-i686	2012-10-01 22:51:28.247181613 +0300
@@ -38,8 +38,8 @@
 bandwidth_limit = 0
 destinationdirectory = /
 distribution = Pardus
-distribution_release = 2011
-distribution_id = p11
+distribution_release = 2013
+distribution_id = p13
 # ftp_proxy = None
 # http_proxy = None
 # https_proxy = None
diff -ur xpisi-2.4/pisi.conf-x86_64 pisi-2.4/pisi.conf-x86_64
--- pisi.conf-x86_64	2012-10-01 22:51:06.000000000 +0300
+++ pisi.conf-x86_64	2012-10-01 22:51:43.117995704 +0300
@@ -38,8 +38,8 @@
 bandwidth_limit = 0
 destinationdirectory = /
 distribution = Pardus
-distribution_release = 2011
-distribution_id = p11
+distribution_release = 2013
+distribution_id = p13
 # ftp_proxy = None
 # http_proxy = None
 # https_proxy = None
