--- redeclipse.sh.orig	2011-11-14 09:25:54.000000000 +0200
+++ redeclipse.sh	2012-01-07 14:13:46.829000002 +0200
@@ -3,10 +3,10 @@
 #RE_DATA=~/redeclipse
 #RE_DATA=/usr/local/redeclipse
 #RE_DATA=.
-RE_DATA=$(dirname $0)
+RE_DATA=/usr/share/redeclipse
 
 # RE_BIN should refer to the directory in which Red Eclipse executable files are placed.
-RE_BIN=${RE_DATA}/bin
+RE_BIN=/usr/bin
 
 # RE_OPTIONS contains any command line options you would like to start Red Eclipse with.
 RE_OPTIONS="-r"
@@ -56,10 +56,10 @@
     MACHINE_SUFFIX=""
 fi
 
-if [ -x ${RE_BIN}/reclient${SYSTEM_SUFFIX}${MACHINE_SUFFIX} ]
+if [ -x ${RE_BIN}/reclient ]
 then
     cd ${RE_DATA}
-    exec ${RE_BIN}/reclient${SYSTEM_SUFFIX}${MACHINE_SUFFIX} ${RE_OPTIONS} "$@"
+    exec ${RE_BIN}/reclient ${RE_OPTIONS} "$@"
 else
     echo "Your platform does not have a pre-compiled Red Eclipse client."
     echo -n "Would you like to build one now? [Yn] "
@@ -76,4 +76,3 @@
         exit 1
     fi
 fi
-
