--- gtkmathview-0.8.0.orig/mathmlsvg/Makefile.in
+++ gtkmathview-0.8.0/mathmlsvg/Makefile.in
@@ -63,7 +63,7 @@
 mathmlsvg_DEPENDENCIES = $(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
 	$(top_builddir)/src/backend/svg/libmathview_backend_svg.la \
 	$(top_builddir)/src/view/libmathview_frontend_libxml2.la \
-	$(am__DEPENDENCIES_1)
+	$(top_builddir)/src/libmathview.la $(am__DEPENDENCIES_1)
 DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(top_builddir)
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
@@ -192,6 +192,7 @@
 PACKAGE_NAME = @PACKAGE_NAME@
 PACKAGE_STRING = @PACKAGE_STRING@
 PACKAGE_TARNAME = @PACKAGE_TARNAME@
+PACKAGE_URL = @PACKAGE_URL@
 PACKAGE_VERSION = @PACKAGE_VERSION@
 PANGOX_CFLAGS = @PANGOX_CFLAGS@
 PANGOX_LIBS = @PANGOX_LIBS@
@@ -285,6 +286,8 @@
   $(GLIB_LIBS) \
   $(top_builddir)/src/backend/svg/libmathview_backend_svg.la \
   $(top_builddir)/src/view/libmathview_frontend_libxml2.la \
+  $(top_builddir)/src/libmathview.la \
+  -lxml2 \
   $(NULL)

 INCLUDES = \
@@ -318,9 +321,9 @@
 	      exit 1;; \
 	  esac; \
 	done; \
-	echo ' cd $(top_srcdir) && $(AUTOMAKE) --foreign  mathmlsvg/Makefile'; \
+	echo ' cd $(top_srcdir) && $(AUTOMAKE) --gnu  mathmlsvg/Makefile'; \
 	cd $(top_srcdir) && \
-	  $(AUTOMAKE) --foreign  mathmlsvg/Makefile
+	  $(AUTOMAKE) --gnu  mathmlsvg/Makefile
 .PRECIOUS: Makefile
 Makefile: $(srcdir)/Makefile.in $(top_builddir)/config.status
 	@case '$?' in \