--- gtkmathview-0.8.0.orig/mathmlps/Makefile.in
+++ gtkmathview-0.8.0/mathmlps/Makefile.in
@@ -59,7 +59,7 @@
 mathmlps_DEPENDENCIES = $(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
 	$(top_builddir)/src/backend/ps/libmathview_backend_ps.la \
 	$(top_builddir)/src/view/libmathview_frontend_libxml2.la \
-	$(am__DEPENDENCIES_1)
+	$(top_builddir)/src/libmathview.la $(am__DEPENDENCIES_1)
 DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(top_builddir)
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
@@ -188,6 +188,7 @@
 PACKAGE_NAME = @PACKAGE_NAME@
 PACKAGE_STRING = @PACKAGE_STRING@
 PACKAGE_TARNAME = @PACKAGE_TARNAME@
+PACKAGE_URL = @PACKAGE_URL@
 PACKAGE_VERSION = @PACKAGE_VERSION@
 PANGOX_CFLAGS = @PANGOX_CFLAGS@
 PANGOX_LIBS = @PANGOX_LIBS@
@@ -268,6 +269,7 @@
   $(GLIB_LIBS) \
   $(top_builddir)/src/backend/ps/libmathview_backend_ps.la \
   $(top_builddir)/src/view/libmathview_frontend_libxml2.la \
+  $(top_builddir)/src/libmathview.la \
   $(NULL)

 INCLUDES = \
@@ -300,9 +302,9 @@
 	      exit 1;; \
 	  esac; \
 	done; \
-	echo ' cd $(top_srcdir) && $(AUTOMAKE) --foreign  mathmlps/Makefile'; \
+	echo ' cd $(top_srcdir) && $(AUTOMAKE) --gnu  mathmlps/Makefile'; \
 	cd $(top_srcdir) && \
-	  $(AUTOMAKE) --foreign  mathmlps/Makefile
+	  $(AUTOMAKE) --gnu  mathmlps/Makefile
 .PRECIOUS: Makefile
 Makefile: $(srcdir)/Makefile.in $(top_builddir)/config.status
 	@case '$?' in \